docs/en/tutorials/runner.md

```diff
@@ -1,3 +1,523 @@
+
+</details>
+
+<details>
+<summary>Why are some arguments passed as dicts?</summary>
+
+Well, this is related to MMEngine's style. In MMEngine, we provide 2 different styles of runner construction: a) manual construction and b) construction via registry. If you are confused, the following example will give a good illustration:
+
+```python
+from mmengine.model import BaseModel
+from mmengine.runner import Runner
+from mmengine.registry import MODELS # root registry for your custom model
+
+@MODELS.register_module() # decorator for registration
+class MyAwesomeModel(BaseModel): # your custom model
+    def __init__(self, layers=18, activation='silu'):
+        ...
+
+# An example of manual construction
+runner = Runner(
+    model=dict(
+        type='MyAwesomeModel',
+        layers=50,
+        activation='relu'),
+    ...
+)
+
+# An example of construction via registry
+model = MyAwesomeModel(layers=18, activation='relu')
+runner = Runner(
+    model=model,
+    ...
+)
+```
+
+Similar to the above example, most arguments in the runner accepts both 2 types of inputs. They are conceptually equivalent. The difference is, in the former style, the module (passed in as a `dict`) will be built **in the runner when actually needed**, while in the latter style, the module has been built before passed to the runner. The following figure illustrates the core idea of registry: it maintains the mapping between a module's **build method** and its **registry name**. If you want to learn more about the full usage of registry, you are recommended to read [Registry](../advanced_tutorials/registry.md) tutorial.
+
+![Runner Registry Illustration](https://user-images.githubusercontent.com/112053249/199191651-44174d17-0fc5-4443-8d15-76f561ec0585.png)
+
+You might still be confused after the explanation. Why should we let the Runner build modules from dicts? What are the benefits? If you have such questions, then we are proud to answer: "Absolutely - no benefits!" In fact, module construction via registry only works to its best advantage when combined with a configuration file. It is still far from the best practice to write as the above example. We provide it here just to make sure you can read and get used to this writing style, which may facilitate your understanding of the actual best practice we will soon talk about - the configuration file. Stay tuned!
+
+If you as a beginner do not immediately understand, it doesn't matter too much, because **manual construction** is still a good choice especially for small scale development and trial-and-error due to its being IDE friendly. However, you are still expected to read and get used to the writing style via registry, so that you can avoid being unnecessarily confused and puzzled in subsequent tutorials.
```

Comment:
```suggestion
If you as a beginner do not immediately understand, it doesn't matter too much, because **manual construction** is still a good choice, especially for small-scale development and trial-and-error due to its being IDE friendly. However, you are still expected to read and get used to the writing style via registry, so that you can avoid being unnecessarily confused and puzzled in subsequent tutorials.
```